### PR TITLE
Library of Babel

### PR DESCRIPTION
**Pull request by Kurfursten**
The Zinc Saucier crate is still coming. But a year of conflicts, ghosted Kanef PRs in particular that I can't see the diff on, and file organization made it impossible to bring into the world in one go. This PR prepares the way.
* It separates Trader.dm into a variety of new files that are more manageable.

- * Note for code review: nothing has changed code-end on anything that was moved into the new crate files. Therefore, you do not need to review alcatrazfour.dm, cloudnine.dm, yantar.dm, shoaljunk.dm, trader_gear.dm, and most of misc_variety.dm (only the nanodictionary stuff at the end is new)

* It makes a few small tweaks that will support Zinc Saucier (e.g.: creates cricket remains, even though crickets don't exist yet, refactors how apiaries are built)
* Includes the Sea Devil buff that was intended for Zinc Saucier
* Includes the Library of Babel crate of nanodictionaries

🆑 
* rscadd: Traders can now buy nanodictionaries originally intended for the Library of Babel.
* tweak: The Sea Devil now eats filth instead of adult fish.

### CODE REVIEW

**Comment by Eneocho** [comment]
> ghosted Kanef PRs in particular that I can't see the diff on

You can press the "browse the repository at this point in history" button to check the PR'd changes, it's cumbersome as hell but it helps.

**Comment by Eneocho** [comment]
Given that we already have the fishtank cleaner module, perhaps the sea devil should take care of feeding the fish at the cost of eating some eggs every now and then? Something like the sea devil's shed skin flakes working as feed.

**Comment by Kurfursten** [comment]
The main feature of sd is that it automatically plants eggs for you. The buff just removes its obtuse drawback

File: code/modules/trader/trade_gear.dm [hunk 1]
```diff
@@ -105,3 +123,59 @@
...
+
+/obj/item/crackerbox/attackby(obj/item/I, mob/user)
+	if(!status && istype(I, /obj/item/weapon/spacecash) && user.drop_item(I, src))
+		status = TRUE
+		var/obj/item/weapon/spacecash/S = I
+		var/crackers_to_dispense = round((S.worth*S.amount)/get_cash2cracker_rate())
+		playsound(loc, 'sound/items/polaroid2.ogg', 50, 1)
+		if(!crackers_to_dispense)
+			say("Not enough! Never enough!")
+		spawn(3 SECONDS)
```
Comment (Exxion) [inline code/modules/trader/trade_gear.dm:167]: No `return`/`else`?

Comment (Exxion) [reply code/modules/trader/trade_gear.dm:167]: Never mind, was already like this

**Pull request merged**